### PR TITLE
fixed wrong accuracy issue when loading a trained model

### PR DESCRIPTION
**Pull request by PaulCCCCCCH**
## Issue ticket number and link

Tag relevant issues and PRs as follows

- Part of issue #159 .

## What are the changes

> Model weights not loaded properly. Loading a trained model by name prints success message, but the test accuracy is the same as a randomly initialized model.

The model weights were correctly loaded and I was confused about the cause at first. After some struggle it turned out that the bug was very simple: we did not call `model.eval` before evaluation. Failing to do so will mess up things such as dropout and batchnorm layers, which results in bad predictions.

## Checklist before requesting a review

- [x] Double check the diff between my branch and the target branch. Make sure there's nothing unexpected.
- [x] Tag relevant issues and PRs in the first section of this form.
- [x] All CI tests are passed.
- [x] No `package-lock.json` or `package.json` changes are pushed. If you do, add a new section `Dependency Changes` to the form stating what library changes are introduced and why.

### CODE REVIEW

**Review by Leon-Leyang — APPROVED** [review]
Looks good to me.
